pkg/provider/onepasswordsdk/client.go

```diff
@@ -0,0 +1,349 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package onepasswordsdk
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/1password/onepassword-sdk-go"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/kube-openapi/pkg/validation/strfmt"
+
+	esv1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1"
+	"github.com/external-secrets/external-secrets/pkg/utils/metadata"
+)
+
+// ErrKeyNotFound is returned when a key is not found in the 1Password Vaults.
+var ErrKeyNotFound = errors.New("key not found")
+
+type PushSecretMetadataSpec struct {
+	Tags  []string `json:"tags,omitempty"`
+	Vault string   `json:"vault,omitempty"`
+}
+
+// GetSecret returns a single secret from the provider.
+// Follows syntax is used for the ref key: https://developer.1password.com/docs/cli/secret-reference-syntax/
+func (p *Provider) GetSecret(ctx context.Context, ref esv1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	if ref.Version != "" {
+		return nil, errors.New(errVersionNotImplemented)
+	}
+	key := p.constructRefKey(ref.Key)
+	secret, err := p.client.Secrets().Resolve(ctx, key)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(secret), nil
+}
+
+// Close closes the client connection.
+func (p *Provider) Close(_ context.Context) error {
+	return nil
+}
+
+// DeleteSecret Not Implemented.
+func (p *Provider) DeleteSecret(ctx context.Context, ref esv1.PushSecretRemoteRef) error {
+	providerItem, err := p.findItem(ctx, ref.GetRemoteKey())
+	if err != nil {
+		return err
+	}
+
+	providerItem.Fields, err = deleteField(providerItem.Fields, ref.GetProperty())
+	if err != nil {
+		return fmt.Errorf("failed to delete fields: %w", err)
+	}
+
+	// There is a chance that there is an empty item left in the section like this: [{ID: Title:}].
+	if len(providerItem.Sections) == 1 && providerItem.Sections[0].ID == "" && providerItem.Sections[0].Title == "" {
+		providerItem.Sections = nil
+	}
+
+	if len(providerItem.Fields) == 0 && len(providerItem.Files) == 0 && len(providerItem.Sections) == 0 {
+		// Delete the item if there are no fields, files or sections
+		if err = p.client.Items().Delete(ctx, providerItem.VaultID, providerItem.ID); err != nil {
+			return fmt.Errorf("failed to delete item: %w", err)
+		}
+		return nil
+	}
+
+	if _, err = p.client.Items().Put(ctx, providerItem); err != nil {
+		return fmt.Errorf("failed to update item: %w", err)
+	}
+	return nil
+}
+
+func deleteField(fields []onepassword.ItemField, title string) ([]onepassword.ItemField, error) {
+	// This will always iterate over all items,
+	// but it's done to ensure that two fields with the same label
+	// exist resulting in undefined behavior
+	var (
+		found   bool
+		fieldsF = make([]onepassword.ItemField, 0, len(fields))
+	)
+	for _, item := range fields {
+		if item.Title == title {
+			if found {
+				return nil, fmt.Errorf("found multiple labels on item %q", title)
+			}
```

Comment:
I think we should document this limitation on the docs - "in order to use ESO with 1Password SDK, documents must have unique label names"

Reply:
added